tests/services/spacingEditor.tests.ts

```diff
@@ -405,8 +405,8 @@ test("Flat track: get spacing to preceding vehicle 1 step away", t =>
 {
 	const mapMock = setupTrackIterator([ flatTrackPiece ]);
 	const train = createTrain(mapMock, [
-		{ trackProgress: 17, trackLocation: flatTrackPiece.position }, // front car
-		{ trackProgress: 16, trackLocation: flatTrackPiece.position }
+		{ trackProgress: 17, trackLocation: {...flatTrackPiece.position, trackType: flatTrackPiece.type} }, // front car
```

Comment:
```suggestion
		{ trackProgress: 17, trackLocation: flatTrackPiece.toLocation() }, // front car
```
Maybe implementing a `toLocation()` method on `TrackPiece` would make these spreads in the test additions a bit less verbose?